eng/common/pipelines/templates/steps/docs-metadata-release.yml

```diff
@@ -30,14 +71,25 @@ steps:
       Write-Host "This script is not executing on Windows, skipping registry modification."
     }
   displayName: Enable Long Paths if Necessary
-  
-- pwsh: |
-    git clone https://github.com/${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }} ${{ parameters.WorkingDirectory }}/repo
-  displayName: Clone Documentation Repository
-  ignoreLASTEXITCODE: false
+
+- ${{ if not(parameters.SparseCheckoutPaths) }}:
```

Comment:
I definitely like this approach better as it should allow both scenarios to work. It might actually be interesting make this an option in the sparse-checkout template itself if we find more patterns like this.

Reply:
Ah you mean perhaps replacing all our checkouts with calls to some checkout template, and we can have a parameter like `sparse` or `full`? I initially tried doing a sparse checkout with `/*` but that actually doubled the checkout time, hence why I went this route instead.

Reply:
Yes having this split of full vs sparse clone might be a future improvement to consider.